exporter/exporter.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package exporter contains an OpenCensus service exporter.
+package exporter
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"sync"
+	"time"
+
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc"
+
+	"github.com/census-instrumentation/opencensus-proto/gen-go/exporterproto"
+	"github.com/census-instrumentation/opencensus-proto/gen-go/traceproto"
+	"github.com/census-instrumentation/opencensus-service/internal"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+var debug bool
+
+func init() {
+	_, debug = os.LookupEnv("OPENCENSUS_DEBUG")
+}
+
+type Exporter struct {
+	OnError func(err error)
+
+	initOnce sync.Once
+
+	clientMu       sync.Mutex
+	clientEndpoint string
+	spanClient     exporterproto.Export_ExportSpanClient
+}
+
+func (e *Exporter) init() {
+	go func() {
```

Comment:
Do we need to start a background Goroutine? Why not just check at export time?

Reply:
I didn't want to check the file and the changes at every time there is export. There are numerous number of exports happening at any time and we don't want to be too aggressively looking up for the file. Btw, we are lazily starting the goroutine at the export time.

Reply:
You could use the a rate limiter to prevent too much checking.

Reply:
In terms of behavior, it is not going to change anything but complicate the implementation.

What's the problem of running a goroutine? It doesn't have any cost. It is not like allocating an actual OS thread or anything.

Reply:
I know but it will appear in stack dumps etc. In general, I would be surprised if a library starts a background process automatically without any opt-in by the user.